ui/src/app/domains/views/DomainDetails/ResourceRecords/ResourceRecords.tsx

```diff
@@ -14,47 +26,121 @@ const ResourceRecords = ({ id }: Props): JSX.Element | null => {
     domainsSelectors.getById(state, Number(id))
   );
 
+  const isAdmin = useSelector(authSelectors.isAdmin);
+
   if (!domain || !domain.rrsets || domain.rrsets.length === 0) {
     return null;
   }
 
   const headers = [
     {
       content: "Name",
+      sortKey: "name",
     },
     {
       content: "Type",
+      sortKey: "type",
     },
     {
       content: "TTL",
+      sortKey: "ttl",
     },
     {
       content: "Data",
+      sortKey: "data",
     },
     {
       content: "Actions",
+      className: "u-align--right",
     },
   ];
 
   const rows = domain.rrsets.map((resource) => {
+    const nodeType: NodeType = resource.node_type;
```

Comment:
You're getting a strict mode error here because there's a disparity between the `NodeType` type that you've given this variable, and the `resource.node_type` type, which is `number | undefined` as defined in the `DomainResource` type.

A couple of things need to happen. First, `DomainResource.node_type` should use the type `NodeType` and not `number`, because `number` is too broad. Secondly, instead of being an optional parameter, it should actually be of type `NodeType | null` because the api actually explicitly sends `null` instead of not including the parameter at all. Then you shouldn't need to explicitly type `const nodeType` here since it'll be inferred to be `NodeType | null`.

I've had a quick look at the websocket code and it seems like the domain types should look something like this:
``` typescript
export enum RecordType {
  AAAA = "AAAA",
  A = "A",
  // etc
}

export type DomainResource = {
  dnsdata_id: number | null;
  dnsresource_id: number | null;
  name: string;
  node_type: NodeType | null;
  rrdata: string | null;
  rrtype: RecordType;
  system_id: BaseNode["system_id"] | null;
  ttl: number | null;
  user_id: User["id"] | null;
};
```